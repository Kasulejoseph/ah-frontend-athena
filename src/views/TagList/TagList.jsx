import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import "./TagList.scss";
import { getArticles } from "../../actions/articleActions/ArticleActions";

export class TagList extends Component {
  handleClick = e => {
    e.preventDefault();
    const { getArticles } = this.props;
    getArticles(`tag=${e.target.name}`);
  };
  render() {
    getArticles();
    const { tags } = this.props;
    if (tags) {
      return (
        <div>
          {tags.map(tag => (
            <button
              tabIndex={-1}
              type="button"
              name={tag}
              onClick={this.handleClick}
              to={`/articles/?tag=${tag}`}
              key={tag}
              className="badge btn-click badge-pill p-2 z-depth-0 badge-primary h4 ml-1 px-3 view overlay"
            >
              {tag}
            </button>
          ))}
        </div>
      );
    } else {
      return <div />;
    }
  }
}

export default connect(
  null,
  { getArticles }
)(TagList);

TagList.propTypes = {
  getArticles: PropTypes.func,
  tags: PropTypes.shape([])
};

TagList.defaultProps = {
  getArticles: () => {},
  tags: []
};
