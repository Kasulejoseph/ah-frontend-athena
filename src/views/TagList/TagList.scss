.badge {
  font-size: 100%;
}
.btn-click {
  &:active {
    background: #fff !important;
    color: #000000;
  }
}
