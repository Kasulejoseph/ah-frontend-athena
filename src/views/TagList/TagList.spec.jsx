import React from "react";
import { shallow, mount } from "enzyme";
import { TagList } from "./TagList";

const handleClick = jest.fn;
const getArticles = jest.fn;
let tags = ["magic", "angular", "andela"];
const props = {
  handleClick,
  getArticles,
  tags
};

describe("<TagList />", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<TagList {...props} />);
  });

  it("matches snapshort", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call handHover ", () => {
    const instance = wrapper.instance();
    expect(
      instance.handleClick({
        preventDefault: jest.fn,
        getArticles: jest.fn,
        target: { value: "react" }
      })
    );
  });

  it("render empty div", () => {
    const mWrapper = mount(<TagList {...props} />);
    mWrapper.setProps(tags);
    expect(mWrapper.find("button").length).toEqual(3);
  });

  it("render empty div", () => {
    const mWrapper = mount(<TagList />);
    expect(mWrapper.find("button").length).toEqual(0);
  });
});
