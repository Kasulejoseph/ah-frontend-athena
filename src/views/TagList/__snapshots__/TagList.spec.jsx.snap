// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagList /> matches snapshort 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TagList
    getArticles={[Function]}
    handleClick={[Function]}
    tags={
      Array [
        "magic",
        "angular",
        "andela",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="badge btn-click badge-pill p-2 z-depth-0 badge-primary h4 ml-1 px-3 view overlay"
          name="magic"
          onClick={[Function]}
          tabIndex={-1}
          to="/articles/?tag=magic"
          type="button"
        >
          magic
        </button>,
        <button
          className="badge btn-click badge-pill p-2 z-depth-0 badge-primary h4 ml-1 px-3 view overlay"
          name="angular"
          onClick={[Function]}
          tabIndex={-1}
          to="/articles/?tag=angular"
          type="button"
        >
          angular
        </button>,
        <button
          className="badge btn-click badge-pill p-2 z-depth-0 badge-primary h4 ml-1 px-3 view overlay"
          name="andela"
          onClick={[Function]}
          tabIndex={-1}
          to="/articles/?tag=andela"
          type="button"
        >
          andela
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "magic",
        "nodeType": "host",
        "props": Object {
          "children": "magic",
          "className": "badge btn-click badge-pill p-2 z-depth-0 badge-primary h4 ml-1 px-3 view overlay",
          "name": "magic",
          "onClick": [Function],
          "tabIndex": -1,
          "to": "/articles/?tag=magic",
          "type": "button",
        },
        "ref": null,
        "rendered": "magic",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "angular",
        "nodeType": "host",
        "props": Object {
          "children": "angular",
          "className": "badge btn-click badge-pill p-2 z-depth-0 badge-primary h4 ml-1 px-3 view overlay",
          "name": "angular",
          "onClick": [Function],
          "tabIndex": -1,
          "to": "/articles/?tag=angular",
          "type": "button",
        },
        "ref": null,
        "rendered": "angular",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "andela",
        "nodeType": "host",
        "props": Object {
          "children": "andela",
          "className": "badge btn-click badge-pill p-2 z-depth-0 badge-primary h4 ml-1 px-3 view overlay",
          "name": "andela",
          "onClick": [Function],
          "tabIndex": -1,
          "to": "/articles/?tag=andela",
          "type": "button",
        },
        "ref": null,
        "rendered": "andela",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="badge btn-click badge-pill p-2 z-depth-0 badge-primary h4 ml-1 px-3 view overlay"
            name="magic"
            onClick={[Function]}
            tabIndex={-1}
            to="/articles/?tag=magic"
            type="button"
          >
            magic
          </button>,
          <button
            className="badge btn-click badge-pill p-2 z-depth-0 badge-primary h4 ml-1 px-3 view overlay"
            name="angular"
            onClick={[Function]}
            tabIndex={-1}
            to="/articles/?tag=angular"
            type="button"
          >
            angular
          </button>,
          <button
            className="badge btn-click badge-pill p-2 z-depth-0 badge-primary h4 ml-1 px-3 view overlay"
            name="andela"
            onClick={[Function]}
            tabIndex={-1}
            to="/articles/?tag=andela"
            type="button"
          >
            andela
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "magic",
          "nodeType": "host",
          "props": Object {
            "children": "magic",
            "className": "badge btn-click badge-pill p-2 z-depth-0 badge-primary h4 ml-1 px-3 view overlay",
            "name": "magic",
            "onClick": [Function],
            "tabIndex": -1,
            "to": "/articles/?tag=magic",
            "type": "button",
          },
          "ref": null,
          "rendered": "magic",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "angular",
          "nodeType": "host",
          "props": Object {
            "children": "angular",
            "className": "badge btn-click badge-pill p-2 z-depth-0 badge-primary h4 ml-1 px-3 view overlay",
            "name": "angular",
            "onClick": [Function],
            "tabIndex": -1,
            "to": "/articles/?tag=angular",
            "type": "button",
          },
          "ref": null,
          "rendered": "angular",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "andela",
          "nodeType": "host",
          "props": Object {
            "children": "andela",
            "className": "badge btn-click badge-pill p-2 z-depth-0 badge-primary h4 ml-1 px-3 view overlay",
            "name": "andela",
            "onClick": [Function],
            "tabIndex": -1,
            "to": "/articles/?tag=andela",
            "type": "button",
          },
          "ref": null,
          "rendered": "andela",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
